src/Resources/views/CRUD/base_list.html.twig

```diff
@@ -259,13 +259,13 @@ file that was distributed with this source code.
     {# NEXT_MAJOR: Remove |default(filter.option('show_filter')) #}
     {% set displayableFilters = admin.datagrid.filters|filter(filter => filter.showFilter|default(filter.option('show_filter')) is not same as (false)) %}
     {%- if displayableFilters|length %}
-        <ul class="nav navbar-nav navbar-right">
+        <ul class="nav navbar-nav navbar-right" id="filter-list-{{ admin.uniqid() }}" {{ stimulus_controller('filter_list', {}, { 'active': 'active' }, { 'filter': '#filter-container-' ~ admin.uniqid() }) }}>
```

Comment:
I haven't used stimulus that much, but we should add a vendor prefix here so it won't clash wish other "list_filter" controller.

Reply:
Indeed, we should use `sonata-` prefix in the safe way we're prefixing twig function with `sonata_` https://github.com/sonata-project/SonataAdminBundle/blob/4.x/src/Twig/Extension/SonataAdminExtension.php#L43

Reply:
by default controller registration is automatic based on file name, file name `filter_controller.js` matches `filter` controller in app. We can name the file `sonata_filter_controller.js` or use [manual registration](https://stimulus.hotwired.dev/handbook/installing#using-other-build-systems)

```js
import { Application } from '@hotwired/stimulus'

import FilterController from '/controllers/filter'

window.Stimulus = Application.start();
Stimulus.register('sonata-filter', FilterController);
```
if use manual registration, the structure folder could be 
 
```
contollers/
- filter.js
- filter_list.js
- revision.js
- treeview.js
- confirm_exit.js
```

IMHO this looks nicer than using auto registration.

Reply:
Symfony is using a `@symfony/` prefix for all components:
https://symfony.com/bundles/StimulusBundle/current/index.html#with-webpackencorebundle

Following that guide, we should use something like this:
```twig
{{ stimulus_controller('@sonata-project/admin/filter-list', { //...
```

Reply:
yeah, with this approach, using outles will be one big pain https://github.com/hotwired/stimulus/issues/641

Reply:
ok, let's use defaults registration and events as communications between controllers, because to use outlets looks crazy with this approach
```
export default class extends Controller {
  static outlets = ['symfony--ux-chartjs--chart'];
  connect() { this.symfonyUxChartjsChartOutlet.method() }
```

Reply:
For what have worked on my projects, you need to add this to the `package.json`:

```
{
    "name": "@sonata-project/admin-bundle",
    "symfony": {
        "controllers": {
            "filter": {
                "main": "dist/calendar.controller.js",
                "fetch": "lazy",
                "enabled": true,
                "autoimport": {
                    "@sonata-project/admin-bundle/dist/style.css": true
                }
            }
        }   
    }
}
```

Not sure, but the `composer.json` file might also need a tag with the name "symfony-ux"

Reply:
I would go with the simplest solution to have a sonata prefix.

If renaming files give them the prefix, it may be the easiest solution

Reply:
> For what have worked on my projects, you need to add this to the `package.json`:
> 
> ```json
> {
>     "name": "@sonata-project/admin-bundle",
>     "symfony": {
>         "controllers": {
>             "filter": {
>                 "main": "dist/calendar.controller.js",
>                 "fetch": "lazy",
>                 "enabled": true,
>                 "autoimport": {
>                     "@sonata-project/admin-bundle/dist/style.css": true
>                 }
>             }
>         }   
>     }
> }
> ```
> 
> Not sure, but the `composer.json` file might also need a tag with the name "symfony-ux":
> 
> ```json
> {
>     "keywords": [
>         "symfony-ux"
>     ]
> }
> ```

I don't think it will work in our case. If you follow this guide https://symfony.com/doc/current/frontend/create_ux_bundle.html for each controller we must specify entrypoint
but webpack build one file `app.js` in output.

The approach you describe could make a package https://github.com/sonata-project/form-extensions and will would resuable ux package, but curently `form-extensions` trying to access our SonataAdmin app from window https://github.com/sonata-project/form-extensions/blob/2.x/assets/js/app.js#L16 and register controller.

SonataAdmin is the central app because create [stimulus application](https://github.com/sonata-project/SonataAdminBundle/blob/4.x/assets/js/stimulus.js)

Reply:
I think it is better to do manual registration like is shown in the form-extensions. Indeed, the sonatadmin is the central app that creates the main Stimulus objects and let others the ability to register more controllers. That's what I thought it could work to not have to add all the controllers here.